### DISCUSSION TITLE
Can Stockfish evaluate the position in which it has not yet been trained?

### DISCUSSION THREAD

**Discussion by ChessOverflow** [official-stockfish/Stockfish#4648] [Q&A] [answered]
Hello! I have recently learned about neural network engine, but I have little knowledge of how it works. I want to know if Stockfish can correctly evaluate positions that it has not trained yet?

**Comment by cj5716** [comment]
you cannot train a neural network on all positions possible to be reached on a chess board. However, a neural network "learns" from the data it was trained on, so that it can give an accurate evaluation even on positions it was not trained on.
see https://github.com/official-stockfish/Stockfish/wiki/Advanced-topics#classical-versus-nnue-evaluation and https://github.com/glinscott/nnue-pytorch/blob/master/docs/nnue.md

**Comment by dav1312** [comment, marked as answer]
This is called "generalization". It refers to the ability of a model to apply concepts learned during training to new data that was not seen during training.

**Reply by ChessOverflow** [reply]
Do you mean by generalization the same as reinforcement learning?

**Reply by dav1312** [reply]
No

**Comment by ChessOverflow** [comment]
How can I download all positions that have been trained so far?

How can I detect if a position has been trained or not?

**Reply by cj5716** [reply]
Training data here: https://robotmoon.com/nnue-training-data/
